docker-files-for-Gitlab-CI-rust/cross/i686/Dockerfile

```diff
@@ -8,26 +8,36 @@ FROM parity/rust-parity-ethereum-build:xenial as builder
 LABEL maintainer="devops-team@parity.io" \
 	vendor="Parity Technologies" \
 	name="parity/rust-parity-ethereum-build" \
-	description="This is for CI build stage for Parity Ethereum linux binary." \
-	vcs-url="https://gitlab.com/paritytech/scripts/docker-files-for-Gitlab-CI-rust/parity-eth-linux-ci" \
-	url="https://gitlab.com/paritytech"
+	description="Image for CI build stage for Parity Ethereum i386 binary." \
+	url="https://gitlab.com/paritytech/scripts" \
+	vcs-url="scripts/docker-files-for-Gitlab-CI-rust/cross/i686"
 
 # install i386 dependencies and tools
-RUN dpkg --add-architecture i386
-
-RUN apt -y update && \
-		apt install -y \
-		gcc-multilib g++-multilib libudev-dev libudev-dev:i386
-
+RUN set -eux; \
+	dpkg --add-architecture i386; \
+# install dependencies
+	apt-get -y update; \
+	apt install -y \
+		gcc-multilib g++-multilib \
+		libudev-dev libudev-dev:i386; \
+# apt clean up
+	apt-get autoremove -y; \
+	apt-get clean; \
+	rm -rf /var/lib/apt/lists/*; \
 # install i686 toolchain
-RUN rustup target add i686-unknown-linux-gnu
+	rustup target add i686-unknown-linux-gnu
 
 # set cross-compiler ENV
 ENV CC=gcc \
-		CXX=g++ \
-		CARGO_TARGET=i686-unknown-linux-gnu \
-		CARGO_HOME=/parity-ethereum/.cargo/ \
-		HOST_CC=gcc \
-		HOST_CXX=g++ \
-		LDFLAGS="-L/usr/lib32" \
-		CPPFLAGS="-I/usr/include"
+	CXX=g++ \
+	CARGO_TARGET=i686-unknown-linux-gnu \
+	HOST_CC=gcc \
+	HOST_CXX=g++ \
+	LDFLAGS="-L/usr/lib32" \
+	CPPFLAGS="-I/usr/include" \
+# cache handler
+	RUSTC_WRAPPER=sccache \
+	CARGO_HOME=/ci-cache/parity-ethereum/cargo \
+	SCCACHE_DIR=/ci-cache/parity-ethereum/sccache \
```

Comment:
does the caching respect the architecture?

Reply:
These changes are waiting for a new config file in `parity-ethereum/scripts/gitlab/` which will have our current target-linker pairs. 
I'll test before merging.
These parameters respect the setup of caching in all our docker builder images.